### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by aborsu**
Logging and throwing and exception is considered
an anti-pattern by many.
- https://community.oracle.com/docs/DOC-983543#logAndThrow
- http://stackoverflow.com/questions/6639963/why-is-log-and-throw-condsidered-an-anti-pattern

But at the same time, catching an exception too far away from where it
was launched can be problematic if you want to add additional
information to help debugging. Or in other case remove sensitive
infromation before logging.

To solve this I propose that each function is responsible for adding
meta information or removing sensitive data and passes it higher up
the stack without logging.

This avoids unnecessary duplicate logs, ensure that an exception is
logged only if the catcher deems it necessary and it can contain all
the information necessary for debugging and nothing sensitive.

### CODE REVIEW

**Comment by SagaStalin** [comment]
<KGB>5af4e5c5c71418843e9443db82ca794cf9867bb2
>feat(error-handling): Stop logging.
>
>Logging and throwing and exception is considered
>an anti-pattern by many.
>
>* https://community.oracle.com/docs/DOC-983543#logAndThrow
>* http://stackoverflow.com/questions/6639963/
>why-is-log-and-throw-condsidered-an-anti-pattern
>
>But at the same time, catching an exception too far away from where it
>was launched can be problematic if you want to add additional
>information to help debugging. Or in other case remove sensitive
>infromation before logging.
>
>To solve this I propose that each function is responsible for adding
>meta information or removing sensitive data and passes it higher up
>the stack without logging.
>
>This avoids unnecessary duplicate logs, ensure that an exception is
>logged only if the catcher deems it necessary and it can contain all
>the information necessary for debugging and nothing sensitive.

Conform status to `<type>(<scope>): <subject>` pattern
type and scope cannot contain spaces or parenthesisMore info [here](https://github.com/sagacify/komitet-gita-bezopasnosti)

File: src/index.js [hunk 1]
```diff
@@ -52,7 +52,11 @@ const methods = loggerRef => {
         .tap(result =>
           loggerRef.debug(buildLog(`${event}_SUCCESS`, result, meta)))
         .catch(err => {
-          loggerRef.error(buildLog(`${event}_FAIL`, err, meta));
+          loggerRef.debug(buildLog(`${event}_SUCCESS`, err, meta));
```
Comment (fbeuvens) [inline src/index.js]: _SUCCESS -> _FAIL

File: src/index.js [hunk 2]
```diff
@@ -48,18 +47,14 @@ const methods = loggerRef => {
...
-          resolve(result);
-        })
+    logMethods.debug(event, null, meta);
+    return func.apply(this, args)
+        .tap(result =>
+          logMethods.debug(`${event}_SUCCESS`, result, meta))
         .catch(err => {
-          loggerRef.error(`${event}_FAIL`, err, meta);
-          reject(err);
+          logMethods.error(`${event}_FAIL`, err, meta);
```
Comment (aborsu) [inline src/index.js]: Aussi corrigé un petit bug, le loggerRef c'est le logger bunyan donc il gère pas les event, data, meta.

**Comment by satblip** [comment]
Mais, ici, tu ne modifies que les logs des fonctions qui sont "logifié", c'est normal?

**Comment by aborsu** [comment]
@satblip oui, comme on ne les contrôle pas, je ne log plus les erreurs (sauf en debug), mais je rajoute le contexte à l'erreur pour pouvoir le loguer avec le contexte si on décide de le faire plus haut dans la chaine.

**Pull request closed without merging**